autogpt/utils/agent_discovery.py

```
@@ -0,0 +1,287 @@
+
+    Returns (ip, family_str) on success, raises ValueError on
+    failure. Uses the first non-blocked address found.
+    """
+    try:
+        addrs = socket.getaddrinfo(
+            domain, 443, proto=socket.IPPROTO_TCP
+        )
+    except socket.gaierror:
+        raise ValueError(
+            f"domain does not resolve: {domain}"
+        )
```

Ruff (0.15.9)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)